src/nginx_config.py

```diff
@@ -276,23 +189,29 @@ def _listen_args(self, port: int, ipv6: bool, ssl: bool, http2: bool) -> List[st
             args.append("http2")
         return args
 
-    def _servers(self, addresses_by_role: Dict[str, Set[str]]) -> List[Dict[str, Any]]:
+    def _servers(
+        self, addresses_by_role: Dict[str, Set[str]], tls: bool = False
+    ) -> List[Dict[str, Any]]:
         servers = []
         roles = addresses_by_role.keys()
 
-        if TempoRole.distributor.value in roles or TempoRole.all.value in roles:
+        if TempoRole.distributor.value in roles:
             for protocol, port in Tempo.receiver_ports.items():
-                servers.append(self._server(port, protocol.replace("_", "-"), "grpc" in protocol))
-        if TempoRole.query_frontend.value in roles or TempoRole.all.value in roles:
+                servers.append(
+                    self._server(port, protocol.replace("_", "-"), "grpc" in protocol, tls)
+                )
+        if TempoRole.query_frontend.value in roles:
```

Comment:
if we have both a distributor and a query-frontend, will this generate a server config for the same protocol twice? 
one for receiver and one for server?

Can you add some comments to clarify what's going on here? why are we adding receiver ports to a servers config?

Reply:
@mmkay please correct me if I'm wrong.
Those are different set of ports. We're adding receiver ports to a servers config for the charms/workloads pushing traces to those different receivers.

Reply:
Yes, server_ports are tempo_http and tempo_grpc - ones for API access. distributor_ports are the ones with receiver protocol ingesters.

Reply:
Added a comment with that